src/remote/util.py

```diff
@@ -117,17 +118,37 @@ def prepare_shell_command(command: Union[str, Sequence[str]]) -> str:
     return " ".join(result)
 
 
-def ssh(host: str, command: str, dry_run: bool = False, raise_on_error: bool = True):
+def ssh(
+    host: str,
+    command: str,
+    dry_run: bool = False,
+    raise_on_error: bool = True,
+    port_forwarding_config: Optional[PortForwardingConfig] = None,
+):
     """Execute a command remotely using SSH and return it's exit code
 
     :param host: a name of the host that will be used for execution
     :param command: a command to execute
     :param dry_run: log command instead of executing it
     :param raise_on_error: raise an exception is remote execution
-
+    :param port_forwarding_config: configs to enable port forwarding while executing the command.
     :returns: exit code of remote command or 255 if connection didn't go through
     """
-    subprocess_command = ["ssh", "-tKq", "-o", "BatchMode=yes", host, command]
+
+    if port_forwarding_config:
+        subprocess_command = [
+            "ssh",
+            "-tKq",
+            "-o",
+            "BatchMode=yes",
+            "-L",
+            f"{port_forwarding_config.local_port}:localhost:{port_forwarding_config.remote_port}",
+            host,
+            command,
+        ]
+    else:
+        subprocess_command = ["ssh", "-tKq", "-o", "BatchMode=yes", host, command]
```

Comment:
Can we unionize it as much as possible, so it will be easier to edit code later? Something like

```
subprocess_command = ["ssh", "-tKq", "-o", "BatchMode=yes"]
if port_forwarding_config:
    subprocess_command.extend((
            "-L",
            f"{port_forwarding_config.local_port}:localhost:{port_forwarding_config.remote_port}",
    ))
subprocess_command.extend((host, command))
```

Reply:
Done